python/kserve/kserve/protocol/dataplane.py

```diff
@@ -209,6 +254,25 @@ def model_ready(self, model_name: str) -> bool:
         Raises:
             ModelNotFound: exception if model is not found
         """
+
+        # If predictor host is present, then it means this is a transformer,
+        # We should also check the predictor model's health if predictor health check is enabled.
+        if DataPlane.predictor_health_check and DataPlane.predictor_host:
+            if DataPlane.predictor_protocol == PredictorProtocol.GRPC_V2.value:
+                grpc_client = get_grpc_client(DataPlane.predictor_host, DataPlane.predictor_use_ssl)
+                res = await grpc_client.ModelReady(pb.ModelReadyRequest(name=model_name))
+                if not res.ready:
+                    return False
+            elif (DataPlane.predictor_protocol == PredictorProtocol.REST_V1.value or
+                  DataPlane.predictor_protocol == PredictorProtocol.REST_V2.value):
+                res = await get_http_client().get(get_model_ready_endpoint(DataPlane.predictor_host,
+                                                                           DataPlane.predictor_protocol,
+                                                                           DataPlane.predictor_use_ssl,
```

Comment:
The predictor configuration should be already setup on the model object itself.

Reply:
We can enhance the `inference_client.py` to support the liveness and readiness client API